### ISSUE TITLE
azureml-automl-runtime==1.12.0.post1, --disable-pip-version-check) exited with code 1. ERROR: THESE PACKAGES DO NOT MATCH THE HASHES FROM THE REQUIREMENTS FILE

### ISSUE THREAD

**Issue by yeamusic21** [Azure/azure-sdk-for-python#34301]
- **Package Name**: 

azureml-automl-runtime

- **Package Version**: 

1.12.0.post1

- **Operating System**:

Ubuntu 18.04.5 LTS

- **Python Version**: 

3.7.5

**Describe the bug**

org.apache.spark.SparkException: Process List(/local_disk0/pythonVirtualEnvDirs/virtualEnv-redacted/bin/python, /local_disk0/pythonVirtualEnvDirs/virtualEnv-redacted/bin/pip, install, azureml-automl-runtime==1.12.0.post1, --disable-pip-version-check) exited with code 1. ERROR: THESE PACKAGES DO NOT MATCH THE HASHES FROM THE REQUIREMENTS FILE. If you have updated the package versions, please update the hashes. Otherwise, examine the package contents carefully; someone may have tampered with them.

**To Reproduce**
Run the following code on a Databricks 7.3 LTS runtime cluster.

`dbutils.library.installPyPI("numpy", version="1.16.2")
dbutils.library.installPyPI("pandas", version="0.23.4")
dbutils.library.installPyPI("scikit-learn", version="0.20.3")
dbutils.library.installPyPI("xgboost", version="0.90")

dbutils.library.installPyPI("azure-common", version="1.1.25")
dbutils.library.installPyPI("azure-core", version="1.8.1")
dbutils.library.installPyPI("azure-graphrbac", version="0.61.1")
dbutils.library.installPyPI("azure-identity", version="1.2.0")
dbutils.library.installPyPI("azure-mgmt-authorization", version="0.61.0")
dbutils.library.installPyPI("azure-mgmt-containerregistry", version="2.8.0")
dbutils.library.installPyPI("azure-mgmt-core", version="1.2.0")
dbutils.library.installPyPI("azure-mgmt-keyvault", version="7.0.0")
dbutils.library.installPyPI("azure-mgmt-resource", version="10.2.0")
dbutils.library.installPyPI("azure-mgmt-storage", version="16.0.0")
dbutils.library.installPyPI("azureml-automl-core", version="1.12.0")
dbutils.library.installPyPI("azureml-automl-runtime", version="1.12.0.post1")
dbutils.library.installPyPI("azureml-contrib-notebook", version="1.12.0")
dbutils.library.installPyPI("azureml-core", version="1.12.0.post2")
dbutils.library.installPyPI("azureml-dataprep", version="2.0.10")
dbutils.library.installPyPI("azureml-dataprep-native", version="20.1.1")
dbutils.library.installPyPI("azureml-dataset-runtime", version="1.12.0")
dbutils.library.installPyPI("azureml-defaults", version="1.12.0")
dbutils.library.installPyPI("azureml-explain-model", version="1.12.0")
dbutils.library.installPyPI("azureml-interpret", version="1.12.0")
dbutils.library.installPyPI("azureml-model-management-sdk", version="1.0.1b6.post1")
dbutils.library.installPyPI("azureml-pipeline", version="1.12.0")
dbutils.library.installPyPI("azureml-pipeline-core", version="1.12.0")
dbutils.library.installPyPI("azureml-pipeline-steps", version="1.12.0")
dbutils.library.installPyPI("azureml-sdk", version="1.12.0")
dbutils.library.installPyPI("azureml-telemetry", version="1.12.0")
dbutils.library.installPyPI("azureml-train", version="1.12.0")
dbutils.library.installPyPI("azureml-train-automl", version="1.12.0")
dbutils.library.installPyPI("azureml-train-automl-client", version="1.12.0.post2")
dbutils.library.installPyPI("azureml-train-automl-runtime", version="1.12.0.post1")
dbutils.library.installPyPI("azureml-train-core", version="1.12.0")
dbutils.library.installPyPI("azureml-train-restclients-hyperdrive", version="1.12.0")
dbutils.library.installPyPI("azureml-widgets", version="1.12.0")

dbutils.library.restartPython()  # Removes Python state, but some libraries might not work without calling this function`

**Expected behavior**
To run without error.

**Screenshots**
N/A

**Additional context**
This ran for months without issue and then between 02/13/2024, 02:25:23 AM and 02/13/2024, 02:30:33 AM Eastern US Time, this error popped up.

**Comment by pvaneck** [comment]
Hey, is this a recurring issue? I am wondering if this is a transient issue that caused some kind of hiccup (possibly networking)

Does retrying work? If not, can you try the following before installing everything:

```
%pip cache purge
```

Also, something to be wary of is that you are also using an end-of-life version of the databricks runtime cluster, an end-of-life version of python, and fairly old versions of those azureml packages,

**Comment by yeamusic21** [comment]
Retrying does work.  

Doesn't seem like much of a solution though.  If it works day after day, and I'm always using the same environment, shouldn't it continue to work?  Why would it stop working out of the blue?  Is Azure making changes to these old package versions?

**Comment by pvaneck** [comment]
If retrying works, then that indicates some sort of transient fault during the download of the package from PyPI. Quite possible that there was a momentary loss of connectivity or a disk error that might have affected the download in some way, causing the hashes to not match. This could be something from the Databricks side, the PyPI side, or somewhere in between.

Regarding the old package versions, these should not have changed, as I believe PyPI doesn't allow overwriting existing uploads/versions.
